app/src/test/java/org/simple/clinic/editpatient/EditPatientEffectHandlerTest.kt

```diff
@@ -148,6 +161,40 @@ class EditPatientEffectHandlerTest {
     verify(patientRepository).updateAddressForPatient(patient.uuid, patientAddress)
     verify(patientRepository).updatePhoneNumberForPatient(patient.uuid, phoneNumber)
     verify(patientRepository).saveBusinessId(updatedBangladeshNationalId)
+    verify(patientRepository, never()).addIdentifierToPatient(any(), any(), any(), any())
+    verifyNoMoreInteractions(patientRepository)
+    testCase.assertOutgoingEvents(PatientSaved)
+    verifyZeroInteractions(ui)
+  }
+
+  @Test
+  fun `adding an id to an empty Bangladesh id should create a new Business id`() {
+    //given
+    val identifier = bangladeshNationalId.identifier
+
+    whenever(patientRepository.updatePatient(patient)) doReturn Completable.complete()
+    whenever(patientRepository.updateAddressForPatient(patient.uuid, patientAddress)) doReturn Completable.complete()
+    whenever(patientRepository.updatePhoneNumberForPatient(patient.uuid, phoneNumber)) doReturn Completable.complete()
+    whenever(patientRepository.saveBusinessId(bangladeshNationalId)) doReturn Completable.complete()
+    whenever(userSession.loggedInUser()) doReturn (Observable.just(user.toOptional()))
+    whenever(facilityRepository.currentFacility(user)) doReturn (Observable.just(facility))
+    whenever(patientRepository.addIdentifierToPatient(patient.uuid, identifier, user, facility)) doReturn Single.just(bangladeshNationalId)
+
+    //when
+    testCase.dispatch(SavePatientEffect(
+        entry.updateBangladeshNationalId(bangladeshNationalId.identifier.value),
```

Comment:
Since we already have `identifier` as a local variable, do we need to reference this via the `bangladeshNationalId` again?